### PR TITLE
[receiver/couchdb] Initialize component

### PR DESCRIPTION
**Pull request by JonathanWamsley**
**Description:**
This PR adds the config and factory for a Couchdb metric receiver.

**Testing:**
Wrote unit tests for config and factory files.

### CODE REVIEW

File: receiver/couchdbreceiver/config.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	u, parseErr := url.Parse(cfg.Endpoint)
+	if parseErr != nil {
+		err = multierr.Append(err, fmt.Errorf(errInvalidEndpoint.Error(), cfg.Endpoint))
+		return err
+	}
+
+	if u.Hostname() == "" {
+		err = multierr.Append(err, fmt.Errorf(errInvalidHost.Error(), cfg.Endpoint))
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (JonathanWamsley) [reply receiver/couchdbreceiver/config.go]: I see. Removed hostname check.

**Comment by djaglowski** [comment]
@JonathanWamsley Please resolve the conflict in `versions.yaml`. Looks good otherwise

**Comment by JonathanWamsley** [comment]
I rebased my branch onto main to resolve the versions conflict @djaglowski

**Comment by djaglowski** [comment]
@JonathanWamsley please fix `versions.yaml`

**Review by djaglowski — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**